### ISSUE TITLE
feat: Deprecate Named Sharding and Update Lightpush Client API

### ISSUE THREAD

**Issue by chair28980** [waku-org/js-waku#1690]
In the [context of discussions](https://discord.com/channels/1110799176264056863/1166349141102837882/1166670331151388784) around named sharding and peer management for the Waku network, it is proposed to make the following changes to the Waku Network:

1. Deprecate Support for Named Sharding:

- It is recommended to remove support for named sharding.
- Peer management should ensure that only peers supporting sharded topics are connected.

2. Lightpush Protocol Updates:

- The lightpush protocol itself should never be aware of the sharded format of pubsub topics. It is an extension of libp2p's gossipsub.
- On a protocol level, the lightpush service node needs to handle requests on unsupported pubsub topics by returning a clear error code.

3. Lightpush Client API Changes:

- To maintain adherence to the sharded format, the lightpush client API should be updated.
- New proposed format: lightPublish(WakuMessage, ClusterID, ShardNumber)
- For legacy requests on named topics, the API could use a trial and error process to send the request to random service nodes. This can be done using the format: lightPublish(WakuMessage, PubsubTopic)
- A request format that bypasses our peer management/selection: lightPublish(WakuMessage, PubsubTopic, ServiceNodeMultiaddr)

- Users of custom pubsub topics to select a cluster id.
- Use static sharding until the launch of The Waku Network (TWN).

### Acceptance Criteria:

- [ ]  Remove support for named sharding.
- [ ]  Update lightpush protocol to handle unsupported pubsub topics with clear error codes.
- [ ]  Introduce the new lightpush client API as mentioned above.
- [ ]  Update documentation to reflect these changes and guide users accordingly.

**Comment by fryorcraken** [comment]
FYI https://github.com/waku-org/js-waku/issues/1749#issuecomment-1839697359

**Comment by fryorcraken** [comment]
Note that Status is contemplating using named shard so I don't think we can fully deprecate.

**Comment by danisharora099** [comment]
> Note that Status is contemplating using named shard so I don't think we can fully deprecate.

Moving this back to triage. Once we have confirmation in either direction, we can handle this issue as required.

**Comment by jm-clius** [comment]
Indeed. There should be no more use of named sharding in any clients.

**Comment by weboko** [comment]
Up to date checklist:
- remove usage of `DefaultPubsubTopic`;
